### PR TITLE
Fix SHL/SHR/SAR PerfScores

### PR DESCRIPTION
**Pull request by pentp**
While doing diffs for #49585 I noticed that SHR PerfScores were way off: `shr reg, cl` costs were assigned to `shr reg, const` instructions.

+ Tuned `mul/imul` latencies to reflect actual use cases.
+ Fixed some RCR/RCL latencies also.
+ Fix DIV/IDIV latencies also (were mistakenly assigned to throughput).

### CODE REVIEW

File: src/coreclr/jit/emitxarch.cpp [hunk 1]
```diff
@@ -14441,12 +14447,12 @@ emitter::insExecutionCharacteristics emitter::getInsExecutionCharacteristics(ins
         case INS_shl_1:
         case INS_shr_1:
         case INS_sar_1:
-            result.insLatency = PERFSCORE_LATENCY_1C;
```
Comment (briansull) [inline src/coreclr/jit/emitxarch.cpp:14444]: With your change no latency is assign for IF_MRW, IF_SRW and IF_ARW

Comment (pentp) [reply src/coreclr/jit/emitxarch.cpp:14444]: Yes, I was following the example of other similar instructions, but then later saw that it's actually pretty inconsistent: #49647

File: src/coreclr/jit/emitxarch.cpp [hunk 2]
```diff
@@ -14373,29 +14373,28 @@ emitter::insExecutionCharacteristics emitter::getInsExecutionCharacteristics(ins
             if ((id->idOpSize() == EA_8BYTE))
             {
                 result.insThroughput = PERFSCORE_THROUGHPUT_52C;
-                result.insLatency    = PERFSCORE_LATENCY_62C;
+                result.insLatency += PERFSCORE_LATENCY_62C;
```
Comment (briansull) [inline src/coreclr/jit/emitxarch.cpp:14376]: I don't believe that the Memory latencies of the source operand (if present) necessarily sum fort these divide operations.

**Review by sandreenko — APPROVED** [review]
LGTM, Brian would be a better person to sign off but he is oof now.

**Comment by pentp** [comment]
So this can be merged? I don't have write access to the repo.

**Comment by sandreenko** [comment]
> So this can be merged? I don't have write access to the repo.

sure, just wanted to give others a few days to add feedback if they had it.

**Pull request merged**